### ISSUE TITLE
[2.x] `ActionCache` is broken on Windows

### ISSUE THREAD

**Issue by adpi2** [sbt/sbt#7656]
Last time I tried sbt 2.x on Windows I noticed that the `ActionCache` was broken because of symbolic links.

**Comment by eed3si9n** [comment]
I asked ChatGPT about this.

- On Windows, it's possible to make symbolic links for files, however you need to run the application with Administrator privileges.
- On Windows 10 and newer versions, it's possible to create symbolic links without requiring Administrator privileges, if you enable Developer Mode.
- Junctions are a type of hard link for directories and can be created by non-admin users without Developer Mode. They behave like symlinks but only work for directories. Creation is rather involved - https://github.com/Atry/scala-junction.
- Hard links can be used for files and do not require admin privileges. Hard links must be on the same volume. A hard link essentially mirrors the original file, and changes to one reflect on the other, so we probably do not want to use this.
- Copying. While not ideal, another simple workaround is to just copy the file or directory in cases where symlinks or hard links aren't practical. This doesn’t offer the same functionality as a link, but it can be a stopgap solution for non-admin users.

/cc @BillyAutrey

**Comment by BillyAutrey** [comment]
I can see situations where users might not have administrator privileges, due to corporate policies on their laptops. In order, that means:
1. I would say the administrator mode requirement means this is not a great solution for us.
2. Requiring developer mode could create some user confusion/supportability issues. Worth investigating the difficulty here.
3. Junctions are interesting, but I don't know that we want to use that library directly. Worth investigating.
4. Being on the same volume reduces our supportability, in case someone wants to use a different volume as their disk cache.
5. This is our default, worst case scenario.

I will say that I've talked to others, and every path to try and support symlinks in Windows (for things like Bazel) ends in pain.

Perhaps a good solution here is to identify, from options 2 and 3, which is the easiest to use. And then allow users to opt out by copying. Or, alternatively, fall back to copying ourselves.

**Comment by eed3si9n** [comment]
I don't think junction would work for us since mapping directories won't give us the flexibility of naming the blob with hash in CAS, and symlinking it with useful names like `*.jar`.

So what I suggest we should do is:

- Start with creating symbolic link wrapped in try-catch, assuming that the user either has an Administrator privileges or has enabled Developer Mode - <https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development>.
- If we hit `catch` clause, we can:
   - Print out a warning saying we noticed that sbt can't create symlink and they should try to enable Developer Mode.
   - Fall back to copying files.
   - For the duration of the sbt session, copy files so we don't keep try-catching.

**Comment by BillyAutrey** [comment]
Yeah fallback seems okay. In the process of installing git with a windows VM, I noticed that even the Git installer has an option and a landing page for symlinks on windows. I feel like this is a good candidate for an option to set globally for Windows users.

**Comment by eed3si9n** [comment]
I'll try to send a PR for fallback so at least the whole thing would function on Windows.
